From d73acdda185b89c4cb919bd66e8fbba60c28df0f Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Tue, 11 Aug 2026 09:19:03 +0000
Subject: [PATCH] Keyguard: Fix colors of slices not updating on doze

Slices were invisible (black) in doze when using light wallpapers
Introduced in https://github.com/LineageOS/android_frameworks_base/commit/a19e59d717ec6d573c11c7e8277bba3c4de189c2

Change-Id: I06abd8bf2e28655cc9e6d81366fd82a13454ec5a

[Port to A16] KeyguardStatusView(Controller) removed upstream; slice view now bound via KeyguardSliceViewBinder. setDarkAmount passthrough added to KeyguardSliceViewController, driven by KeyguardInteractor.dozeAmount flow (replaces NotificationPanelViewController.onDozeAmountChanged wiring).

Change-Id: Ib61919659dfd1f723417bea682a7f58172144cf1
---
 .../com/android/keyguard/KeyguardSliceViewController.java  | 7 +++++++
 .../systemui/keyguard/ui/binder/KeyguardSliceViewBinder.kt | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardSliceViewController.java b/packages/SystemUI/src/com/android/keyguard/KeyguardSliceViewController.java
index f8bde98014ab..01d19f3603a6 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardSliceViewController.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardSliceViewController.java
@@ -140,6 +140,13 @@ public class KeyguardSliceViewController extends ViewController<KeyguardSliceVie
         startCoroutines(EmptyCoroutineContext.INSTANCE);
     }
 
+    /**
+     * The amount we're in doze.
+     */
+    public void setDarkAmount(float darkAmount) {
+        mView.setDarkAmount(darkAmount);
+    }
+
     void startCoroutines(CoroutineContext context) {
         collectFlow(mView, mKeyguardInteractor.getDozeTimeTick(),
                 (Long millis) -> {
diff --git a/packages/SystemUI/src/com/android/systemui/keyguard/ui/binder/KeyguardSliceViewBinder.kt b/packages/SystemUI/src/com/android/systemui/keyguard/ui/binder/KeyguardSliceViewBinder.kt
index b420373099ac..e276aef88ffd 100644
--- a/packages/SystemUI/src/com/android/systemui/keyguard/ui/binder/KeyguardSliceViewBinder.kt
+++ b/packages/SystemUI/src/com/android/systemui/keyguard/ui/binder/KeyguardSliceViewBinder.kt
@@ -45,6 +45,12 @@ object KeyguardSliceViewBinder {
                     }
                 }
 
+                launch {
+                    keyguardInteractor.dozeAmount.collect {
+                        keyguardSliceViewController.setDarkAmount(it)
+                    }
+                }
+
                 launch {
                     aodBurnInViewModel.movement.collect { burnInModel ->
                         keyguardSliceViewController.translate(
-- 
2.43.0

